File: java-spring/crud/src/main/java/com/example/crud/service/DeleteFeedService.java
package com.example.crud.service;

import com.example.crud.entity.Feed;
import com.example.crud.repository.FeedRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@RequiredArgsConstructor
@Service
public class DeleteFeedService {

    private final FeedRepository feedRepository;

    @Transactional
    public void execute(Long feedId) {

        Feed feed = feedRepository.findById(feedId)
                .orElseThrow(RuntimeException::new);

        feedRepository.delete(feed);
    }
}

File: kotlin-spring/crud/src/main/kotlin/com/example/crud/service/DeleteFeedService.kt
package com.example.crud.service

import com.example.crud.entity.Feed
import com.example.crud.repository.FeedRepository
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class DeleteFeedService(
        private val feedRepository: FeedRepository,
) {

    @Transactional
    fun execute(feedId: Long) {
        val feed: Feed = feedRepository.findByIdOrNull(feedId) ?: throw RuntimeException();
        feedRepository.delete(feed)
    }
}